genai-perf/genai_perf/config/generate/perf_analyzer_config.py

```
@@ -0,0 +1,146 @@
+            "request-rate": "--request-rate-range",
+        }
+
+        return obj_to_cli_dict[objective_name]
+
+    ###########################################################################
+    # Representation Methods
+    ###########################################################################
+    def representation(self) -> str:
+        """
+        A string representation of the PA command that removes values which
+        can vary between runs, but should be ignored when determining
+        if a previous (checkpointed) run can be used
+        """
+        options_with_arg_to_remove = [
+            "--url",
+            "--metrics-url",
+            "--latency-report-file",
+            "--measurement-request-count",
+        ]
+        options_only_to_remove = ["--verbose", "--extra-verbose", "--verbose-csv"]
+
+        representation = self.create_cli_string()
+
+        # Remove the PA call path which is always the first token
+        representation_list = representation.split(" ")
+        representation_list.pop(0)
+        representation = " ".join(representation_list)
+
+        for option_with_arg in options_with_arg_to_remove:
+            representation = self._remove_option_from_cli_string(
+                option_with_arg, representation, with_arg=True
+            )
+
+        for option_only in options_only_to_remove:
+            representation = self._remove_option_from_cli_string(
+                option_only, representation, with_arg=False
+            )
+
+        return representation
+
+    def _remove_option_from_cli_string(
+        self, option_to_remove: str, cli_string: str, with_arg: bool
+    ) -> str:
+        cli_str_tokens = cli_string.split(" ")
+
+        try:
+            removal_index = cli_str_tokens.index(option_to_remove)
+            cli_str_tokens.pop(removal_index)
+
+            if with_arg:
+                cli_str_tokens.pop(removal_index)
+        except ValueError:
```

What is this meant to catch? Could this cause a silent error?

---

It's so if the option we are trying to remove doesn't exist in the string we can just ignore and return the string unmodified.
I've updated the comment to better reflect this.

---

Would you be able to do this without a try-except by checking if the index is invalid first? And in the with_arg case, checking the index is valid before calling pop (or calling another function that doesn't cause an error if the index is invalid)?

I think try/excepts are best to avoid when possible (e.g. they are sometimes necessary for external code or non-deterministic scenarios). They can obfuscate errors and the except blocks are computationally expensive when executed.

---

Not a problem. I've replaced the try-except with a check to see if the substring is present.

---

Great work!